.github/workflows/main.yaml

```diff
@@ -129,6 +146,39 @@ jobs:
           body_path: ./release-artifacts/changelog.md
           draft: ${{ startsWith(steps.get_tag.outputs.git_tag, 'nightly') != true }}
           prerelease: ${{ startsWith(steps.get_tag.outputs.git_tag, 'nightly') }}
+  push-image:
+    name: Push Image
+    needs:
+      [create-release]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          push: true
+          tags: |
+            kubent:${{ github.sha }},
+            kubent:${{ needs.create-release.outputs.tag_name }}
+      - name: Push latest
+        uses: docker/build-push-action@v3
+        if: ${{ !contains(github.ref, 'nightly') }}
```

Comment:
I think this should work 👍 , can we just remove the `${{`? It should not be needed for ifs [^1] and is consistent with other uses + a bit more readable.

[^1]: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif:~:text=create%20a%20conditional.-,When%20you%20use%20expressions%20in%20an%20if%20conditional%2C%20you%20may%20omit%20the%20expression%20syntax%20(%24%7B%7B%20%7D%7D)%20because%20GitHub%20automatically%20evaluates%20the%20if%20conditional%20as%20an%20expression.%20For%20more%20information%2C%20see%20%22Expressions.%22,-Example%3A%20Only%20run

Reply:
its needed for the !

Reply:
Ah, that makes sense. I think that's however a YAML, rather than GH, issue. I.e. it would need to be quoted, as `!` at the beginning denotes tag.